scripts/check_site_lib.lua

```diff
@@ -103,3 +103,26 @@ function need_string_array(varname, required)
    return assert(pcall(need_array, varname, function(e) assert_type(e, 'string') end, required),
 		 "site.conf error: expected `" .. varname .. "' to be a string array")
 end
+
+function need_var_in_array(varname, array, required)
```

Comment:
Most of this function could be replaced by a call to `need_array`, similar to the way `need_string_array` is defined. I think it would make sense to split this function into two - one that just checks if a value is one of list of possible values (maybe `need_one_of`), and one that works like `need_string_array` and calls `need_array` with `need_one_of` (could be called `need_array_of`).